dispatch/filter.go

```diff
@@ -0,0 +1,243 @@
+
+		default:
+			f.queueOverflow()
+			f.Measures.DroppedQueueCount.Add(1.0)
+		}
+	}
+}
+
+// called if queue is filled
+func (f Filter) queueOverflow() {
+
+	f.Mutex.Lock()
+	if time.Now().Before(f.DropUntil) {
+		f.Mutex.Unlock()
+		return
+	}
+	f.DropUntil = time.Now().Add(f.CutOffPeriod)
+	f.Measures.DropUntilGauge.Set(float64(f.DropUntil.Unix()))
+	secret := f.Norn.Destination.HttpConfig.Secret
+	failureMsg := f.FailureMsg
+	failureURL := f.Norn.Destination.HttpConfig.FailureURL
+	f.Mutex.Unlock()
+
+	var (
+		errorLog = log.WithPrefix(f.Logger, level.Key(), level.ErrorValue())
+	)
+
+	f.Measures.CutOffCounter.Add(1.0)
+
+	// We empty the queue but don't close the channel, because we're not
+	// shutting down.
+	f.empty(f.Measures.DroppedCutoffCounter)
+
+	msg, err := json.Marshal(failureMsg)
+	if nil != err {
+		errorLog.Log(logging.MessageKey(), "Cut-off notification json.Marshal failed", "failureMessage", f.FailureMsg,
+			"for", f.Norn.Destination.HttpConfig.URL, logging.ErrorKey(), err)
+		return
+	}
+
+	// if no URL to send cut off notification to, do nothing
+	if "" == failureURL {
+		return
+	}
+
+	// Send a "you've been cut off" warning message
+	payload := bytes.NewReader(msg)
+	req, err := http.NewRequest("POST", failureURL, payload)
+	if nil != err {
+		// Failure
+		errorLog.Log(logging.MessageKey(), "Unable to send cut-off notification", "notification",
+			failureURL, "for", f.Norn.Destination.HttpConfig.URL, logging.ErrorKey(), err)
+		return
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	if "" != secret {
+		h := hmac.New(sha1.New, []byte(secret))
+		h.Write(msg)
+		sig := fmt.Sprintf("sha1=%s", hex.EncodeToString(h.Sum(nil)))
+		req.Header.Set("X-Codex-Signature", sig)
+	}
+
+	//  record content type, json.
+	f.Measures.ContentTypeCounter.With("content_type", "json").Add(1.0)
+	resp, err := f.Sender(req)
+	if nil != err {
+		// Failure
+		errorLog.Log(logging.MessageKey(), "Unable to send cut-off notification", "notification",
+			failureURL, "for", f.Norn.Destination.HttpConfig.URL, logging.ErrorKey(), err)
+		return
+	}
+
+	if nil == resp {
+		// Failure
+		errorLog.Log(logging.MessageKey(), "Unable to send cut-off notification, nil response",
+			"notification", failureURL)
+		return
+	}
+
+}
+
+func (f *Filter) empty(droppedCounter metrics.Counter) {
+	droppedMsgs := f.FilterQueue.Load().(chan *wrp.Message)
+	f.FilterQueue.Store(make(chan *wrp.Message, f.QueueSize))
+	droppedCounter.Add(float64(len(droppedMsgs)))
+	f.Measures.EventQueueDepthGauge.Set(0.0)
+	return
+}
+
+func (f Filter) sendEvents() {
+	defer f.Wg.Done()
+	queue := f.FilterQueue.Load().(chan *wrp.Message)
+	select {
+
+	case en, ok := <-queue:
+		if !ok {
+			break
+		}
+		f.Measures.EventQueueDepthGauge.Add(-1.0)
+
+		f.Mutex.RLock()
+		deliverUntil := time.Unix(0, f.Norn.ExpiresAt)
+		dropUntil := f.DropUntil
+		f.Mutex.RUnlock()
+
+		now := time.Now()
+
+		if now.Before(dropUntil) {
+			f.Measures.DroppedCutoffCounter.Add(1.0)
```

Comment:
```suggestion
			f.Measures.DroppedCutoffCounter.Add(1.0)
			continue
```
if we don't continue to the next event, we will send this event when we shouldn't.